app/scripts/controllers/edit/deploymentConfig.js

```diff
@@ -0,0 +1,297 @@
+            $scope.pullSecrets = $scope.deploymentConfig.spec.template.spec.imagePullSecrets || [{name: ''}];
+            $scope.volumeNames = _.map($scope.deploymentConfig.spec.template.spec.volumes, 'name');
+            $scope.strategyData = angular.copy($scope.deploymentConfig.spec.strategy);
+            $scope.originalStrategy = $scope.strategyData.type;
+            $scope.strategyParamsPropertyName = getParamsPropertyName($scope.strategyData.type);
+
+            // If strategy is 'Custom' and no environment variables are present, initiliaze them.
+            if ($scope.strategyData.type === 'Custom' && !_.has($scope.strategyData, 'customParams.environment')) {
+              $scope.strategyData.customParams.environment = [];
+            }
+            
+            DataService.list("secrets", context, function(secrets) {
+              var secretsByType = SecretsService.groupSecretsByType(secrets);
+              // Add empty option to the image/source secrets
+              $scope.secretsByType = _.each(secretsByType, function(secretsArray) {
+                secretsArray.unshift("");
+              });
+            });
+
+            // If we found the item successfully, watch for changes on it
+            watches.push(DataService.watchObject("deploymentconfigs", $routeParams.deploymentconfig, context, function(deploymentConfig, action) {
+              if (action === 'MODIFIED') {
+                $scope.alerts["updated/deleted"] = {
+                  type: "warning",
+                  message: "This deployment configuration has changed since you started editing it. You'll need to copy any changes you've made and edit again."
+                };
+              }
+              if (action === "DELETED") {
+                $scope.alerts["updated/deleted"] = {
+                  type: "warning",
+                  message: "This deployment configuration has been deleted."
+                };
+                $scope.disableInputs = true;
+              }
+              $scope.deploymentConfig = deploymentConfig;
+            }));
+            $scope.loaded = true;
+          },
+          // failure
+          function(e) {
+            $scope.loaded = true;
+            $scope.alerts["load"] = {
+              type: "error",
+              message: "The deployment configuration details could not be loaded.",
+              details: $filter('getErrorDetails')(e)
+            };
+          }
+        );
+      })
+    );
+
+    // helper for detemining if strategy switch was done between Rolling <-> Recreate strategy
+    var isRollingRecreateSwitch = function() {
+      return ($scope.strategyData.type !== 'Custom' && $scope.originalStrategy !== 'Custom' && $scope.strategyData.type !== $scope.originalStrategy);
+    };
+
+    $scope.strategyChanged = function() {
+      var pickedStrategyParams = getParamsPropertyName($scope.strategyData.type);
+      if (isRollingRecreateSwitch()) {
```

Comment:
I'd pull everything inside this `if` into a helper function, maybe `promptToMoveParams`. It'd be easier to follow the if/else logic in strategyChanged.